cmd/test_test.go

```diff
@@ -400,7 +401,13 @@ func TestWatchMode(t *testing.T) {
 			<-done
 		}()
 
-		time.Sleep(500 * time.Millisecond)
+		expected := "Watching for changes ..."
+		if !test.Eventually(t, 2*time.Second, func() bool {
```

Comment:
Solves case `1`, where the tests allows the watcher to print multiple outputs for a single file update.
Also has the benefit of being fast when possible, while allowing for greater margins for the host to be slow.